vector/src/main/java/im/vector/util/MediaAVScanner.java

```diff
@@ -56,140 +40,128 @@ public MediaAVScanner(String userId) {
-                ObjectInputStream ois = new ObjectInputStream(fis);
-                mMediaAVScanByUrl = (HashMap) ois.readObject();
-                ois.close();
-                fis.close();
+        }, new Realm.Transaction.OnSuccess() {
+            @Override
+            public void onSuccess() {
+                realm.close();
             }
-        } catch (Exception e) {
-            Log.e(LOG_TAG, "## openMediaAVScannerDict failed " + e.getMessage());
-        }
+        }, new Realm.Transaction.OnError() {
+            @Override
+            public void onError(Throwable error) {
+                Log.e(LOG_TAG, "## clearAntivirusScanResultsCache : Realm transaction failed");
+            }
+        });
     }
 
     /**
      * Get the current scan status for a media.
      * Trigger a scan if it is not already done.
      *
-     * @param context   the context.
      * @param url       the media url.
      * @param callback  optional async response handler.
      * @return the current scan status.
      */
-    public ScanStatus scanEncryptedMedia(Context context, String url, @Nullable ApiCallback<Void> callback) {
-        if (null == mMediaAVScanByUrl) {
-            openMediaAVScannerDict(context);
-        }
+    public MediaAntivirusScanStatus.ScanStatus scanMedia(String url, @Nullable ApiCallback<Void> callback) {
 
         if (TextUtils.isEmpty(url)) {
-            return ScanStatus.UNKNOWN;
+            return MediaAntivirusScanStatus.ScanStatus.UNKNOWN;
         }
 
-        if (mMediaAVScanByUrl.containsKey(url)) {
-            return mMediaAVScanByUrl.get(url);
+        getMediaAvScanByUrl(url);
+
+        if (null != getMediaAvScanByUrl(url)) {
+            return getMediaAvScanByUrl(url);
         }
 
+        return MediaAntivirusScanStatus.ScanStatus.IN_PROGRESS;
+
         // TODO trigger the scan, use the callback on result if any
 
-        return ScanStatus.IN_PROGRESS;
     }
 
     /**
      * Get the current scan status for an encrypted media.
      * Trigger a scan if it is not already done.
      *
-     * @param context the context.
      * @param mediaInfo  the encrypted media information.
      * @param callback   optional async response handler.
      * @return the current scan status.
      */
-    public ScanStatus scanEncryptedMedia(Context context, EncryptedFileInfo mediaInfo, @Nullable ApiCallback<Void> callback) {
-        if (null == mMediaAVScanByUrl) {
-            openMediaAVScannerDict(context);
-        }
+    public MediaAntivirusScanStatus.ScanStatus scanEncryptedMedia(EncryptedFileInfo mediaInfo, @Nullable ApiCallback<Void> callback) {
 
         if (TextUtils.isEmpty(mediaInfo.url)) {
-            return ScanStatus.UNKNOWN;
+            return MediaAntivirusScanStatus.ScanStatus.UNKNOWN;
         }
 
-        if (mMediaAVScanByUrl.containsKey(mediaInfo.url)) {
-            return mMediaAVScanByUrl.get(mediaInfo.url);
+        getMediaAvScanByUrl(mediaInfo.url);
```

Comment:
3 times the same call. Please optimize

Reply:
In the end, all messages will be encrypted and there will be only the scanEncryptedMedia method that will call the method getMediaAvScanByUrl. The scanMedia method will be deleted.